alembic/versions/3a677cb88b01_create_algorithm_tables.py

```diff
@@ -0,0 +1,53 @@
+"""create algorithm tables
+
+Revision ID: 3a677cb88b01
+Revises: 6052c193a26a
+Create Date: 2024-09-19 17:07:41.379176
+
+"""
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision: str = '3a677cb88b01'
+down_revision: Union[str, None] = '6052c193a26a'
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('algorithm',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('is_default', sa.Boolean(), nullable=False),
```

Comment:
Should there be a constraint that ensures only 1 row in the `algorithm` table has `is_default` value of `true`?

Reply:
[Partial indexes are cool](https://www.postgresql.org/docs/current/indexes-partial.html), and hopefully one day is becomes standard ANSI SQL, but to my knowledge Postgresql is the only RDMS we are planning on supporting that has this feature.  Alternatively, Cason added a function in models.py that should prevent users from making this mistake.

Reply:
I had it as a constraint initially but it would have needed to be changed for any other type of DB. There's a function that is run whenever an `update` or `insert` happens on the algorithm table that should take care of this.